onnxruntime/core/framework/tensor_type_and_shape.h

```diff
@@ -6,6 +6,9 @@ struct OrtTensorTypeAndShapeInfo {
  public:
   ONNXTensorElementDataType type = ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT;
   onnxruntime::TensorShape shape;
+  // dim_param values. empty string if dim_value or no dim_param was specified.
+  // one entry per dimension in shape. only guaranteed to be populated for graph inputs and outputs
+  std::vector<std::string> dim_params;
```

Comment:
Should this be part of TensorShape instead so that it can be used in other places as well?

Reply:
I considered it but I don't think so. Adding copies of strings to all TensorShape instances has a good chance of causing performance issues. We have seen perf improvements from simply avoiding the copy of the std::vector of dims in the ctor.

Reply:
Ok. I was assuming that most of the time you would've symbolic dims in the input tensors only and this vector would be empty in all other cases.